File: app/src/main/java/com/example/Sleepy/adapters/WakeCardsAdapter.java
package com.example.Sleepy.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chauthai.swipereveallayout.SwipeRevealLayout;
import com.chauthai.swipereveallayout.ViewBinderHelper;
import com.example.Sleepy.R;
import com.example.Sleepy.shared.MyAlarm;

import java.util.List;

public class WakeCardsAdapter extends RecyclerView.Adapter<WakeCardsAdapter.ViewHolder> {
    private final List<WakeCards> alarmCards;
    View view;

    private final ViewBinderHelper viewBinderHelper = new ViewBinderHelper();

    public WakeCardsAdapter(List<WakeCards> cards) {
        this.alarmCards = cards;
    }

    @NonNull
    @Override
    public WakeCardsAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_item_wake, parent, false);
        return new WakeCardsAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(WakeCardsAdapter.ViewHolder holder, int position) {
        WakeCards state = alarmCards.get(position);
        holder.titleView.setText(state.getTime());
        holder.remTime.setText(state.getRemainingTime());

        viewBinderHelper.setOpenOnlyOne(true);
        viewBinderHelper.bind(holder.srlCard, state.getTime());

        holder.ivAddAlarm.setOnClickListener(view1 -> MyAlarm.setAlarm(view.getContext(), state.getTriggerTime(), view1));
        //Toast.makeText(view.getContext(), "" + new SimpleDateFormat("HH:mm", Locale.getDefault()).format(state.getTriggerTime()), Toast.LENGTH_SHORT).show());
    }

    @Override
    public int getItemCount() {
        return alarmCards.size();
    }

    public void delete(int position){
        alarmCards.remove(position);
        notifyItemRemoved(position);
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        final TextView titleView, remTime;
        final ImageView ivAddAlarm;
        final SwipeRevealLayout srlCard;

        ViewHolder(View view){
            super(view);
            titleView = view.findViewById(R.id.tvTitleCard);
            remTime = view.findViewById(R.id.tvSecCard);
            ivAddAlarm = view.findViewById(R.id.ibAddAlarm);
            srlCard = view.findViewById(R.id.srlCard);
        }
    }
}

File: app/src/main/java/com/example/sleepy/presentation/wake/recycler/WakeCardsAdapter.kt
package com.example.sleepy.presentation.wake.recycler

import androidx.recyclerview.widget.RecyclerView
import com.chauthai.swipereveallayout.ViewBinderHelper
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.example.sleepy.R
import android.widget.TextView
import com.chauthai.swipereveallayout.SwipeRevealLayout
import com.example.sleepy.utils.AlarmUtils
import com.example.sleepy.utils.AnimationsUtils

class WakeCardsAdapter(private val alarmCards: ArrayList<WakeCards>) :
    RecyclerView.Adapter<WakeCardsAdapter.ViewHolder>() {
    private lateinit var view: View
    private val viewBinderHelper = ViewBinderHelper()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        view = LayoutInflater.from(parent.context).inflate(R.layout.card_item_wake, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val state = alarmCards[position]
        holder.titleView.text = state.time
        holder.remTime.text = state.remainingTime
        viewBinderHelper.setOpenOnlyOne(true)
        viewBinderHelper.bind(holder.srlCard, state.time)

        holder.ivAddAlarm.setOnClickListener { view1: View? ->
            AlarmUtils.setAlarm(view.context, state.getTriggerTime(), view1!!)
        }

        AnimationsUtils.setFadeAnimationStart(holder.itemView)

        if(position == 5){
            holder.isOptimal.visibility = View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return alarmCards.size
    }

    fun delete(position: Int) {
        alarmCards.removeAt(position)
        notifyItemRemoved(position)
    }

    class ViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(
        view
    ) {
        val titleView: TextView = view.findViewById(R.id.tvTitleCard)
        val remTime: TextView = view.findViewById(R.id.tvSecCard)
        val ivAddAlarm: ImageView = view.findViewById(R.id.ibAddAlarm)
        val srlCard: SwipeRevealLayout = view.findViewById(R.id.srlCard)
        val isOptimal: ImageView = view.findViewById(R.id.ivOptimalTime)
    }
}